### PR TITLE
tesseract4: apply patches to fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by Artturin**
it wasn't tagged but it does exist
https://github.com/tesseract-ocr/tesseract/blob/4.1/ChangeLog

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
> it wasn't tagged but it does exist

Please open an upstream issue asking them to add a tag.

**Comment by Artturin** [comment]
> > it wasn't tagged but it does exist
> 
> Please open an upstream issue asking them to add a tag.

https://github.com/tesseract-ocr/tesseract/issues/3638

**Comment by dotlambda** [comment]
Can people from @NixOS/darwin-maintainers test this and `ocrmypdf` on Intel and M1?

**Comment by dotlambda** [comment]
> it wasn't tagged but it does exist https://github.com/tesseract-ocr/tesseract/blob/4.1/ChangeLog

I don't think that date is a release date, so maybe 4.1.2 actually hasn't been released yet?

**Comment by Artturin** [comment]
Idk what's going on with their releases https://github.com/tesseract-ocr/tesseract/issues/3331#issuecomment-922260229

Maybe I'll just changes this to unstable----

**Comment by NilsIrl** [comment]
I'm happy to say it builds on an M1 Mac, which means this would fix https://github.com/NixOS/nixpkgs/issues/143059

**Comment by prusnak** [comment]
From https://github.com/tesseract-ocr/tesseract/issues/3638#issuecomment-967453751:

> Release 4.1.2 is prepared in the ChangeLog, but not finished. So currently there is no 4.1.2 release.

**Comment by dotlambda** [comment]
I think we should rather apply https://github.com/tesseract-ocr/tesseract/commit/6dc4b184b1ebf2e68461f6b63f63a033bc7245f7 and https://github.com/tesseract-ocr/tesseract/commit/dbc79b09d195490dfa3f7d338eadac07ad6683f7 as patches.

**Review by dotlambda — COMMENTED** [review]
Please add a comment linking to https://github.com/tesseract-ocr/tesseract/issues/3447

File: pkgs/applications/graphics/tesseract/tesseract4.nix [hunk 1]
```diff
@@ -12,6 +12,17 @@ stdenv.mkDerivation rec {
     sha256 = "1ca27zbjpx35nxh9fha410z3jskwyj06i5hqiqdc08s2d7kdivwn";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/tesseract-ocr/tesseract/commit/dbc79b09d195490dfa3f7d338eadac07ad6683f7.patch";
+      sha256 = "sha256-lGlg0etuU4RXfdq1QH2bYObdeGrFHKf9O8zMUAbfNIQ=";
+    })
```
Comment (NilsIrl) [inline pkgs/applications/graphics/tesseract/tesseract4.nix:19]: This patch is not necessary for tesseract4 to build. Other than that, it works perfectly (with and without this first patch).

**Comment by domenkozar** [comment]
@GrahamcOfBorg build tesseract4

**Comment by stweil** [comment]
Tesseract release 4.1.3 is now available, so you can use a tagged version.

**Comment by xave** [comment]
Is there any update here? I'm trying to use `ocrmypdf` which relies on tesseract. I was the spawn for this series of update activity.

**Comment by dotlambda** [comment]
@xave Does ocrmypdf not work on aarch64-darwin?

**Comment by xave** [comment]
It does not. 

```
nix upgrade-nix
nix-env -i ocrmypdf
```

This then gives the standard:

a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.

**Comment by dotlambda** [comment]
@xave You should do `nix-channel --update`: https://nixos.org/manual/nix/stable/package-management/channels.html

**Comment by xave** [comment]
That worked. Thank you!

**Pull request merged**